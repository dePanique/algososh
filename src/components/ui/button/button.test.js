import renderer from 'react-test-renderer';
import { Button } from './button';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

it('should render Button without text', () => {
    const component = renderer.create(
        <Button 
        />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('should render Button with text', () => {
    const component = renderer.create(
        <Button 
            text='Hello world'
        />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('should render Button with disabled props', () => {
    const component = renderer.create(
        <Button 
            disabled
        />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('should render Button with isLoader props', () => {
    const component = renderer.create(
        <Button 
            isLoader
        />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('should called callback once onClick', async () => {
    const handleClick = jest.fn();

    const user = userEvent.setup();

    render(
            <Button 
                text='Button'
                onClick={handleClick}
                data-testid="testingButton"
            />
    )

    await user.click(screen.getByTestId("testingButton"));

    expect(handleClick).toHaveBeenCalledTimes(1);
})

