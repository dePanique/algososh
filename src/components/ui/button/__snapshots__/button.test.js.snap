// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button with disabled props 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`should render Button with isLoader props 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`should render Button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Hello world
  </p>
</button>
`;

exports[`should render Button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
